### PR TITLE
Add PlacementAPI to ControlPlane CR

### PR DESCRIPTION
**Pull request by dprince**

### CODE REVIEW

File: pkg/openstack/placement.go [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+// ReconcilePlacementAPI -
+func ReconcilePlacementAPI(ctx context.Context, instance *corev1beta1.OpenStackControlPlane, helper *helper.Helper) (ctrl.Result, error) {
+	placementAPI := &placementv1.PlacementAPI{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "placement",
+			Namespace: instance.Namespace,
+		},
+	}
+
+	helper.GetLogger().Info("Reconciling placementAPI", "placementAPI.Namespace", instance.Namespace, "placementAPI.Name", "keystone")
```
Comment (abays) [inline pkg/openstack/placement.go:26]: I think `"placementAPI.Name"` shouldn't have double-quotes around it here

Comment (abays) [reply pkg/openstack/placement.go:26]: Nevermind, I see what you're doing

Comment (abays) [inline pkg/openstack/placement.go:26]: But `s/keystone/placement`

Comment (gibizer) [inline pkg/openstack/placement.go:26]: Based on the discussion in https://coreos.slack.com/archives/CQXJFGMK6/p1663595278584359 we should use the logging wrappers from https://github.com/openstack-k8s-operators/lib-common/blob/master/modules/common/util/log.go But probably that is for a separate patch to replace all the helper.GetLogger() calls in the openstack-operator.

File: pkg/openstack/placement.go [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+	op, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), placementAPI, func() error {
+		instance.Spec.PlacementTemplate.DeepCopyInto(&placementAPI.Spec)
+		if placementAPI.Spec.Secret == "" {
+			placementAPI.Spec.Secret = instance.Spec.Secret
+		}
+		if placementAPI.Spec.NodeSelector == nil && instance.Spec.NodeSelector != nil {
+			placementAPI.Spec.NodeSelector = instance.Spec.NodeSelector
+		}
+		if placementAPI.Spec.DatabaseInstance == "" {
+			placementAPI.Spec.DatabaseInstance = "openstack" //FIXME: see above
```
Comment (gibizer) [inline pkg/openstack/placement.go:36]: The FIXME probably need to refer to the comment in https://github.com/openstack-k8s-operators/openstack-operator/blob/becaa3f96db8e694e0a0dbec91a73d2ab89afac0/pkg/openstack/keystone.go#L36

File: controllers/core/openstackcontrolplane_controller.go [hunk 3]
```diff
@@ -140,6 +149,7 @@ func (r *OpenStackControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) err
 		For(&corev1beta1.OpenStackControlPlane{}).
 		Owns(&mariadbv1.MariaDB{}).
 		Owns(&keystonev1.KeystoneAPI{}).
+		Owns(&placementv1.PlacementAPI{}).
```
Comment (gibizer) [inline controllers/core/openstackcontrolplane_controller.go:152]: Just out of curiosity does this line means
 - a) the openstack-operator controller watches the PlacementAPI CRs *it creates*
 - b) the openstack-operator controller watches *all* PlacementAPI CR creation even if it is created manually outside of the openstack-operator controller.

Comment (abays) [reply controllers/core/openstackcontrolplane_controller.go:152]: It would be `b`.  But it is possible to only have it watch specific `PlacementAPI` CRs, if they are properly labeled and then filtered through a watch function, like we do for ConfigMaps here: https://github.com/openstack-k8s-operators/cinder-operator/blob/77b8278c88f1327ff4ccc050f360a33d246b4c47/controllers/cinderapi_controller.go#L199-L247

Comment (gibizer) [reply controllers/core/openstackcontrolplane_controller.go:152]: Thanks @abays

File: pkg/openstack/placement.go (hunk 1 above)
Comment (dprince) [reply pkg/openstack/placement.go:26]: agreed

File: pkg/openstack/placement.go (hunk 2 above)
Comment (dprince) [reply pkg/openstack/placement.go:36]: I think we can remove it for now

**Comment by gibizer** [comment]
looks good to me

**Review by abays — APPROVED** [review]
/lgtm
